src/store/store.ts

```diff
@@ -0,0 +1,19 @@
+import { create } from "zustand";
+import { devtools, persist, createJSONStorage } from "zustand/middleware";
+import { PTOStore } from "./types";
+import { getToken } from "./functions";
+
+const usePTOStore = create<PTOStore>()(
+  persist(
+    devtools((set) => ({
+      token: undefined,
+      setToken: (token: string) => set({ token }),
+    })),
+    {
+      name: "pto-store",
+      storage: createJSONStorage(() => sessionStorage),
```

Comment:
Just curious: Does this work like using the `Window.sessionStorage` out of the box?

Reply:
it does but I have removed the storage from the frontend and cached the token on the pto-api instead.